### ISSUE TITLE
How can I disable bell sound?

### ISSUE THREAD

**Issue by hwooo** [dbcli/mycli#954]
I'm using mycli on Windows Terminal.
I disabled bell sound in the terminal so there is no beep on cmd, ps, and original MySQL cli.

But there is still a beep sound inside mycli. Is there any option to disable bell inside of mycli?

**Comment by j-bennet** [comment]
mycli is using python-prompt-toolkit to handle console input/output. Starting from v3.0.14, python-prompt-toolkit lets you disable bell using an environment variable:

https://github.com/prompt-toolkit/python-prompt-toolkit/blob/7b28b254aba3452b4ac3a797942474d14c92561f/CHANGELOG#L41

Give it a try and let us know if this helps.

**Comment by hwooo** [comment]
Do I have to install python-prompt-toolkit separately? or can I configure it in .myclirc file?

**Comment by j-bennet** [comment]
If you have mycli installed, you already have prompt-toolkit installed. Mycli would not work without it.

`python -m pip freeze` would show you all installed python modules and their versions.

**Comment by hwooo** [comment]
I can't find proper solution without rebuilding mycli :(

**Comment by j-bennet** [comment]
I guess I should have asked how you installed mycli in the first place. It was not `pip install mycli`?

**Comment by hwooo** [comment]
No, I intalled mycli with pip.

**Comment by j-bennet** [comment]
So what does `pip freeze | grep prompt-toolkit` tell you?

**Comment by hwooo** [comment]
In PowerShell, `pip freeze | findstr prompt-toolkit` gives me following results:
```
prompt-toolkit @ file:///tmp/build/80754af9/prompt-toolkit_1602688806899/work
```

**Comment by j-bennet** [comment]
I see. Not very helpful. How about this?

``
$ python
>>> import prompt_toolkit
>>> prompt_toolkit.__version__
'3.0.8'
```

If you have 3.0.14 or above, running mycli like this should work:

```
PROMPT_TOOLKIT_BELL=false mycli
```

If you have something lower, you can upgrade prompt-toolkit with:

```
pip install -U prompt-toolkit
```

alternatively, you could upgrade mycli and all related packages with:

```
pip install -U mycli
```

see if this helps.

**Comment by hwooo** [comment]
```
❯ PROMPT_TOOLKIT_BELL=false mycli
PROMPT_TOOLKIT_BELL=false : The term 'PROMPT_TOOLKIT_BELL=false' is not recognized as the name of a cmdlet, function,
script file, or operable program.
```
I have `prompt_toolkit` of version 3.0.16, but it doesn't seem to work on Windows.

**Comment by j-bennet** [comment]
You're right, Windows / PowerShell has different syntax. It would be something like this:

```
> $env:PROMPT_TOOLKIT_BELL = 'false'
> mycli
```

I'm on a Mac, so I'm flying blind here, sorry.

**Comment by hwooo** [comment]
It's okay. But `$env:PROMPT_TOOLKIT_BELL = 'false'` also doesn't work. 😂

**Comment by j-bennet** [comment]
I see. I should give up on trying to fix a Windows problem! @dbcli/mycli-core are any of you able to try this on Windows?

**Comment by rolandwalker** [comment]
Closing since we don't test on Windows, and Windows support is best-effort only.
